system-test/datastore.ts

```diff
@@ -15,41 +15,39 @@
  */
 
 import * as assert from 'assert';
-import * as async from 'async';
 import {Datastore} from '../src';
 import {entity} from '../src/entity';
 
 describe('Datastore', () => {
   const testKinds: Array<{}> = [];
   const datastore = new Datastore();
   // Override the Key method so we can track what keys are created during the
-    // tests. They are then deleted in the `after` hook.
+  // tests. Keys are then deleted in the `after` hook.
   const key = datastore.key;
   datastore.key = function(options) {
     const keyObject = key.call(this, options);
     testKinds.push(keyObject.kind);
     return keyObject;
   };
 
-  after(done => {
-    function deleteEntities(kind, callback) {
-      const query = datastore.createQuery(kind).select('__key__');
+  after(async () => {
+    async function deleteEntities(kind) {
+      const query = await datastore.createQuery(kind).select('__key__');
 
-      datastore.runQuery(query, (err, entities) => {
-        if (err) {
-          callback(err);
-          return;
-        }
-
-        const keys = entities.map(entity => {
-          return entity[datastore.KEY];
-        });
-
-        datastore.delete(keys, callback);
+      const data = await datastore.runQuery(query);
+      const entities = data[0];
+      const keys = entities.map(entity => {
+        return entity[datastore.KEY];
       });
+
+      await datastore.delete(keys);
     }
 
-    async.each(testKinds, deleteEntities, done);
+    let promises: Array<{}> = [];
```

Comment:
No need for the loop. 

`const promises = testKinds.map(kind => deleteEntities(kind))`